aws/resource_aws_dms_event_subscription_test.go

```diff
@@ -0,0 +1,247 @@
+	     source_endpoint_arn       = "${aws_dms_endpoint.dms_endpoint_source.endpoint_arn}"
+	     table_mappings            = "{\"rules\":[{\"rule-type\":\"selection\",\"rule-id\":\"1\",\"rule-name\":\"1\",\"object-locator\":{\"schema-name\":\"%%\",\"table-name\":\"%%\"},\"rule-action\":\"include\"}]}"
+
+	     tags = {
+	       Name   = "tf-test-dmses-%[1]s"
+	       Update = "to-update"
+	       Remove = "to-remove"
+	     }
+
+	     target_endpoint_arn = "${aws_dms_endpoint.dms_endpoint_target.endpoint_arn}"
+	   }
+
+	   resource "aws_sns_topic" "topic" {
+	     name = "tf-test-dmses-%[1]s"
+	   }
+	   `, randId)
+}
+
+func dmsEventSubscriptionConfig(randId string) string {
+	return fmt.Sprintf(`
+
+resource "aws_dms_event_subscription" "dms_event_subscription" {
+  name             = "tf-test-dmses-%[1]s"
+  enabled          = true
+  event_categories = ["creation", "failure"]
+  source_type      = "replication-task"
```

Comment:
To simplify testing, it may be easier to pare the base configuration and this down to just `replication-instance` as the source type.